packages/doenetml-worker-rust/lib-doenetml-core/tests/test_utils.rs

```diff
@@ -0,0 +1,173 @@
+//! This file contains utilities for testing DoenetMLCore.
+use assert_json_diff::assert_json_eq;
+use serde_json;
+pub use serde_json::{json, Value};
+use std::process::Command;
+use std::str;
+
+/// Execute the command `node ./tests/dist/parse-dast.js -i <dast>`.
+/// This returns a JSON string of the parsed DAST. Node must be installed and properly configured.
+///
+/// If `strip_position` is true, the position field will be stripped from the output.
+pub fn evaluate_dast_via_node(dast: &str, strip_position: bool) -> std::io::Result<Value> {
+    evaluate_dast_via_node_to_json(dast, strip_position)
+        .map(|val| serde_json::from_str(&val).unwrap())
+}
+
+/// Execute the command `node ./tests/dist/parse-dast.js -i <dast>`.
+/// This returns a JSON string of the parsed DAST. Node must be installed and properly configured.
+///
+/// If `strip_position` is true, the position field will be stripped from the output.
+pub fn evaluate_dast_via_node_to_json(dast: &str, strip_position: bool) -> std::io::Result<String> {
+    let mut command = Command::new("node");
+    command
+        .arg("./tests/dist/parse-dast.js")
+        .arg("-i")
+        .arg(dast);
+    if strip_position {
+        command.arg("--strip-position");
+    }
+    let output = command.output()?;
+
+    match output.status.success() {
+        true => Ok(str::from_utf8(&output.stdout).unwrap().to_string()),
+        false => Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            str::from_utf8(&output.stderr).unwrap().to_string(),
+        )),
+    }
+}
+
+/// Expand the string containing DoenetML into a JSON string of the parsed DAST
+/// omitting the position props.
+#[macro_export]
+macro_rules! dast_no_position {
```

Comment:
Just curious: what do these macros do that couldn't be done with a function?

Reply:
Good point! Theoretically there could be more flexibility down the line, but I have converted them all to functions.